### PR TITLE
planner: separate cost model ver1/ver2 into different files (part 2)

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #36243

Problem Summary:

### What is changed and how it works?

planner: separate cost model ver1/ver2 into different files (part 2)

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/plan_cost_ver2_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+			"select /*+ use_index(t, primary), stream_agg() */ sum(a), sum(a+2), sum(a+4) from t group by b",
+		},
+		{ // hash join uses the small table to build hash table
+			"select /*+ hash_join_build(t1) */ * from t t1, t t2 where t1.b=t2.b and t1.a<10",
+			"select /*+ hash_join_build(t2) */ * from t t1, t t2 where t1.b=t2.b and t1.a<10",
+		},
+		{ // hash join more join keys
+			"select /*+ hash_join_build(t1) */ * from t t1, t t2 where t1.b=t2.b",
+			"select /*+ hash_join_build(t1) */ * from t t1, t t2 where t1.a=t2.a and t1.b=t2.b",
+		},
```
Comment (qw4990) [inline planner/core/plan_cost_ver2_test.go:122]: I'll add more cases in the next several PRs.

**Review by Reminiscent — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]

**Pull request merged**